drivers/ufs/ufs.c

```diff
@@ -1442,8 +1456,42 @@ static void prepare_prdt_table(struct ufs_hba *hba, struct scsi_cmd *pccb)
 
 	prepare_prdt_desc(&prd_table[table_length - i - 1], buf, datalen - 1);
 
+	flush_dcache_range((uintptr_t)pccb->pdata, (uintptr_t)pccb->pdata + pccb->datalen);
 	req_desc->prd_table_length = table_length;
 }
+/*
+ * ufshcd_get_rsp_upiu_data_seg_len - Get the data segment length
+ *				from response UPIU
+ * @ucd_rsp_ptr: pointer to response UPIU
+ *
+ * Return the data segment length.
+ */
+static inline unsigned int
+ufshcd_get_rsp_upiu_data_seg_len(struct utp_upiu_rsp *ucd_rsp_ptr)
+{
+	return be32_to_cpu(ucd_rsp_ptr->header.dword_2) &
+		MASK_RSP_UPIU_DATA_SEG_LEN;
+}
+
+/**
+ * ufshcd_copy_sense_data - Copy sense data in case of check condition
+ * @lrbp: pointer to local reference block
+ */
+static inline void ufshcd_copy_sense_data(struct utp_upiu_rsp *ucd_rsp_ptr,
+					  u8 *sense_buffer)
+{
+	int len;
+
+	if (sense_buffer &&
+	    ufshcd_get_rsp_upiu_data_seg_len(ucd_rsp_ptr)) {
+		int len_to_copy;
+
+		len = be16_to_cpu(ucd_rsp_ptr->sr.sense_data_len);
+		len_to_copy = min_t(int, RESPONSE_UPIU_SENSE_DATA_LENGTH, len);
+
+		memcpy(sense_buffer, ucd_rsp_ptr->sr.sense_data, len_to_copy);
+	}
+}
```

Comment:
Code of `ufshcd_get_rsp_upiu_data_seg_len()` & `ufshcd_copy_sense_data()` routines not needed, remove it from this patch